File: app/src/main/java/com/ruideraj/secretelephant/PropertiesReader.java
package com.ruideraj.secretelephant;

import android.content.Context;
import android.content.res.AssetManager;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

public class PropertiesReader {

    public static String getProperty(Context context, String key) {
        Properties properties = new Properties();
        AssetManager assetManager = context.getAssets();
        try {
            InputStream inputStream = assetManager.open(context.getString(R.string.test_filename));
            properties.load(inputStream);
        }
        catch(IOException io) {
            io.printStackTrace();
        }

        return properties.getProperty(key);
    }

}

File: app/src/main/java/com/ruideraj/secretelephant/PropertiesReader.kt
package com.ruideraj.secretelephant

import android.content.Context
import java.io.IOException
import java.util.*
import javax.inject.Inject

interface PropertiesReader {
    fun getProperty(stringId: Int): String
    fun getProperty(key: String): String
}

class PropertiesReaderImpl @Inject constructor(private val context : Context) : PropertiesReader {

    override fun getProperty(stringId: Int) = getProperty(context.getString(stringId))

    override fun getProperty(key : String) : String {
        val properties = Properties()
        val assetManager = context.assets

        try {
            val inputStream = assetManager.open(context.getString(R.string.test_filename))
            properties.load(inputStream)
        }
        catch(io : IOException) {
            io.printStackTrace()
        }

        return properties.getProperty(key)
    }

}